lighthouse/lighthouse-auth.mjs

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable no-console */
+
+import fs from 'fs';
+
+/**
+ * @param {puppeteer.Page} page
+ * @param {string} origin
+ */
+async function login(page, origin) {
+  if (process.env.PREVIEW_URL) {
+    await loadInitialPage(page, origin);
+  } else {
+    await page.goto(origin);
+  }
+
+  let signInButton;
+  try {
+    signInButton = await page.$('#sign-in-button');
+    if (signInButton === null) {
+      return;
+    }
+  } catch (error) {
+    console.log(' Did not find signInButton', error);
+    return;
+  }
+
+  // Redirect to Okta login
+  const redirectToOktaPromise = page.waitForNavigation();
+  await signInButton.click();
+  await redirectToOktaPromise;
+
+  // Fill in and submit login form.
+  try {
+    await loginToOkta(page);
+  } catch (error) {
+    // Sometimes the login to Okta fails for some reason and puts us back on the /login page of MPDx, so try again.
+    try {
+      signInButton = await page.$('#sign-in-button', { timeout: 5000 });
+
+      if (signInButton) {
+        const redirectToOktaPromise = page.waitForNavigation();
+        await signInButton.click();
+        await redirectToOktaPromise;
+        try {
+          await loginToOkta(page);
+        } catch (loginError) {
+          console.log('Failed to login', loginError);
+        }
+      }
+    } catch (error) {
+      console.log(' Did not find signInButton', error);
+      return;
+    }
+  }
+}
+
+export async function loadInitialPage(page, origin, timeout = 60_000) {
```

Comment:
Uggh, this waiting approach of waiting on the status code to not be 404 probably won't work. I just made a PR with [a preview environment](https://menu-component-error.d3dytjb8adxkk5.amplifyapp.com), and while the first build is building, the response to any page is 200 with the following body:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Welcome</title>
    <style>
        body {
            background-color: #d5dbdb;
            margin: 5em 5em 5em 5em;
            font-family: "Amazon Ember", Helvetica, Arial, sans-serif;
        }

        div {
            background-color: #ffffff;
            padding: 1em 2em 2em 2em;
        }

        h1 {
            color: #16191f;
        }

        h3 {
            color: #16191f;
        }
    </style>
</head>
<body>
<div>
    <h1>Welcome</h1>
    <h3>Your app will appear here once you complete your first deployment.</h3>
    <p>
        Deployment didn't work? Here are some options:
        <ul>
            <li>Check out <b><a href="https://docs.aws.amazon.com/amplify/latest/userguide/welcome.html">our docs</a></b></li>
            <li>Click the <b>Feedback</b> button in the bottom-left corner of the service page</li>
        </ul>
    </p>
    <p>
        Quick tips:
        <ul>
            <li>Have you checked your build settings? The <b>baseDirectory</b> parameter in the 
                <b>artifacts</b> step of your YAML file should match your build output directory</li>
            <li>Building your app should produce an <b>index.html</b> file. Try building your app 
                locally and check a file with that name exists in the artifacts base directory.</li>
        </ul>
        </p>
</div>
</body>
</html>
```

We may have to make this method try again if the page content contains `Your app will appear here once you complete your first deployment.` and hope Amplify doesn't change that placeholder page's content.

Reply:
Or we could cycle wait on something on all of our pages, if such a thing exists.

Reply:
Looks like I could look for `<div id="__next">` which seems to be consistent.

Reply:
I've made a commit to this effect, let me know what you think.